### ISSUE TITLE
New WhatsApp version errors

### ISSUE THREAD

**Issue by justrach** [residentsummer/watoi#38]
Here are a few errors that come when I transfer my database(>>4GB) over using the tool
```
CoreData: warning: Unable to load class named 'WAChatSession' for entity 'WAChatSession'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupInfo' for entity 'WAGroupInfo'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupMember' for entity 'WAGroupMember'.  Class not found, using default NSManagedObject instead.
```

And 

```
2021-03-27 20:16:52.439 watoi[33199:173729] -[NSNull doubleValue]: unrecognized selector sent to instance 0x1f30eb780
2021-03-27 20:16:52.439 watoi[33199:173729] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull doubleValue]: unrecognized selector sent to instance 0x1f30eb780'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000018d61a320 __exceptionPreprocess + 240
	1   libobjc.A.dylib                     0x000000018d348c04 objc_exception_throw + 60
	2   CoreFoundation                      0x000000018d6a9020 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x000000018d57c184 ___forwarding___ + 1444
	4   CoreFoundation                      0x000000018d57bb30 _CF_forwarding_prep_0 + 96
	5   watoi                               0x000000010220cc0c -[Importer convertAndroidTimestamp:] + 36
	6   watoi                               0x000000010220d3f0 -[Importer importChats] + 1312
	7   watoi                               0x000000010220c2a4 -[Importer import] + 28
	8   watoi                               0x000000010220bdcc main + 364
	9   libdyld.dylib                       0x000000018d4bdf34 start + 4
)
libc++abi.dylib: terminating with uncaught exception of type NSException
zsh: abort      build/Release/watoi /Volumes/T7/msgstore.db ./ChatStorage.sqlite 
```

Is there anything that I am doing wrong. Thanks for your assistance

**Comment by residentsummer** [comment]
Hi, @justrach !

In the last line of your log I see the command that you've used to run the utility:

```
build/Release/watoi /Volumes/T7/msgstore.db ./ChatStorage.sqlite
```

If it is what you've run, than path to db model files (`.momd`) is missing.

**Comment by justrach** [comment]
The code that was used to run the utility was as follows:

`(base) poops-MBA-M1 watoi % build/Release/watoi /Volumes/T7/msgstore.db  ./ChatStorage.sqlite app/Payload/WhatsApp.app/Frameworks/Core.framework/WhatsAppChat.momd
2021-03-30 17:58:47.737 watoi[56885:848204] Probing msgstore.db at path: /Volumes/T7/msgstore.db
2021-03-30 17:58:47.737 watoi[56885:848204]     ok.
2021-03-30 17:58:47.739 watoi[56885:848204] Android store loaded
2021-03-30 17:58:47.739 watoi[56885:848204] Probing WhatsAppChat.momd at path: app/Payload/WhatsApp.app/Frameworks/Core.framework/WhatsAppChat.momd
2021-03-30 17:58:47.739 watoi[56885:848204]     ok.
2021-03-30 17:58:47.739 watoi[56885:848204] Probing ChatStorage.sqlite at path: ./ChatStorage.sqlite
2021-03-30 17:58:47.739 watoi[56885:848204]     ok.
2021-03-30 17:58:47.749 watoi[56885:848204] CoreData loaded
CoreData: warning: Unable to load class named 'WAChatSession' for entity 'WAChatSession'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupInfo' for entity 'WAGroupInfo'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupMember' for entity 'WAGroupMember'.  Class not found, using default NSManagedObject instead.
`

This is then followed by several other numbers which cannot be found according to the program

**Comment by residentsummer** [comment]
Command seems right, maybe it just got truncated in the first logs.

`CoreData: warning: Unable to load class ` warnings are "expected".

Looking at the stack trace from your first message, it seems that there is a chat in your android  database, that is not present in iOS backup (A) and does not have a valid creation time set (B).

Not sure why (A) has happened. (B) is likely due to android db format change. I'll need to get my hands on some recent `msgtore.db` to inspect it...

**Comment by justrach** [comment]
> Command seems right, maybe it just got truncated in the first logs.
> 
> `CoreData: warning: Unable to load class ` warnings are "expected".
> 
> Looking at the stack trace from your first message, it seems that there is a chat in your android database, that is not present in iOS backup (A) and does not have a valid creation time set (B).
> 
> Not sure why (A) has happened. (B) is likely due to android db format change. I'll need to get my hands on some recent `msgtore.db` to inspect it...

Hihi, I think the new db structure has to do with WhatsApp potentially enabling cross platform usage, as seen in their source code :)

**Comment by ludufre** [comment]
This PR works. https://github.com/residentsummer/watoi/pull/47

**Comment by residentsummer** [comment]
Should be fixed now by #47. @justrach feel free to reopen if it's not.
